sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategySuite.scala

```diff
@@ -0,0 +1,307 @@
+      DataSourceStrategy.translateFilter(
+        expressions.GreaterThan(1, attrInt))
+    }
+
+    assertResult(Some(sources.GreaterThanOrEqual("cint", 1))) {
+      DataSourceStrategy.translateFilter(
+        expressions.GreaterThanOrEqual(attrInt, 1))
+    }
+    assertResult(Some(sources.GreaterThanOrEqual("cint", 1))) {
+      DataSourceStrategy.translateFilter(
+        expressions.LessThanOrEqual(1, attrInt))
+    }
+
+    assertResult(Some(sources.LessThanOrEqual("cint", 1))) {
+      DataSourceStrategy.translateFilter(
+        expressions.LessThanOrEqual(attrInt, 1))
+    }
+    assertResult(Some(sources.LessThanOrEqual("cint", 1))) {
+      DataSourceStrategy.translateFilter(
+        expressions.GreaterThanOrEqual(1, attrInt))
+    }
+
+    assertResult(Some(sources.In("cint", Array(1, 2, 3)))) {
+      DataSourceStrategy.translateFilter(
+        expressions.InSet(attrInt, Set(1, 2, 3)))
+    }
+
+    assertResult(Some(sources.In("cint", Array(1, 2, 3)))) {
+      DataSourceStrategy.translateFilter(
+        expressions.In(attrInt, Seq(1, 2, 3)))
+    }
+
+    assertResult(Some(sources.IsNull("cint"))) {
+      DataSourceStrategy.translateFilter(
+        expressions.IsNull(attrInt))
+    }
+    assertResult(Some(sources.IsNotNull("cint"))) {
+      DataSourceStrategy.translateFilter(
+        expressions.IsNotNull(attrInt))
+    }
+
+    assertResult(Some(sources.And(
+      sources.GreaterThan("cint", 1),
+      sources.LessThan("cint", 10)))) {
+      DataSourceStrategy.translateFilter(expressions.And(
+        expressions.GreaterThan(attrInt, 1),
+        expressions.LessThan(attrInt, 10)
+      ))
+    }
+
+    assertResult(Some(sources.Or(
+      sources.GreaterThanOrEqual("cint", 8),
+      sources.LessThanOrEqual("cint", 2)))) {
+      DataSourceStrategy.translateFilter(expressions.Or(
+        expressions.GreaterThanOrEqual(attrInt, 8),
+        expressions.LessThanOrEqual(attrInt, 2)
+      ))
+    }
+
+    assertResult(Some(sources.Not(
+      sources.GreaterThanOrEqual("cint", 8)))) {
+      DataSourceStrategy.translateFilter(
+        expressions.Not(expressions.GreaterThanOrEqual(attrInt, 8)
+        ))
+    }
+
+    assertResult(Some(sources.StringStartsWith("cstr", "a"))) {
+      DataSourceStrategy.translateFilter(
+        expressions.StartsWith(attrStr, "a"
+        ))
+    }
+
+    assertResult(Some(sources.StringEndsWith("cstr", "a"))) {
+      DataSourceStrategy.translateFilter(
+        expressions.EndsWith(attrStr, "a"
+        ))
+    }
+
+    assertResult(Some(sources.StringContains("cstr", "a"))) {
+      DataSourceStrategy.translateFilter(
+        expressions.Contains(attrStr, "a"
+        ))
+    }
+  }
+
+  test("translate complex expression") {
+    val attrInt = 'cint.int
+    // Functions such as 'Abs' are not supported
+    assertResult(None) {
+      DataSourceStrategy.translateFilter(
+        expressions.LessThanOrEqual(
+          expressions.Subtract(expressions.Abs(attrInt), 2), 1))
```

Comment:
can we move the comment to this line? i.e.
```
// `Abs` expression cannot be pushed down
expressions.Subtract(expressions.Abs(attrInt), 2), 1))
```

Reply:
Fixed. Thanks.